sound/soc/sof/ipc4-topology.c

```diff
@@ -736,10 +743,61 @@ static int sof_ipc4_widget_setup_comp_mixer(struct snd_sof_widget *swidget)
 err:
 	sof_ipc4_free_audio_fmt(&mixer->available_fmt);
 	kfree(mixer);
+	return ret;
```

Comment:
@RanderWang im not sure what happened here but we need to keep the swidget->private = NULL; for the mixer too

Reply:
sure. Let me create another PR to address it